### PR TITLE
Reuse VMs as part of compass-integration jobs

### PR DESCRIPTION
**Pull request by dzahariev**
**Description**
Reuse VMs as part of compass-integration jobs

Changes proposed in this pull request:
- Add PJTester and increase tracing
- Introduce pre and post jobs
- Refactor job scripts

Replaces https://github.com/kyma-project/test-infra/pull/6045

### CODE REVIEW

File: vpath/pjtester.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+pjNames:
```
Comment (NickyMateev) [inline vpath/pjtester.yaml:1]: Remove this file before merge

File: prow/scripts/provision-vm-compass.sh [hunk 2]
```diff
@@ -16,31 +16,69 @@ source "${SCRIPT_DIR}/lib/utils.sh"
...
+# Download yq
+YQ_VERSION="v4.25.1"
+log::info "Downloading yq version: $YQ_VERSION"
+wget "https://github.com/mikefarah/yq/releases/download/${YQ_VERSION}/yq_linux_amd64"
+mv yq_linux_amd64 yq
+chmod +x yq
+
+# Flag to recycle or recreate VM on each commit execution
+RECYCLE_VM="no"
+PROVISIONING_PASS="no"
```
Comment (NickyMateev) [inline prow/scripts/provision-vm-compass.sh:28]: Can you put a more meaningful name than `PROVISIONING_PASS` as it doesn't tell me anything just looking at it AND/OR add a comment explaining it like with the `RECYCLE_VM`.

**Review by PetarTodorovv — CHANGES_REQUESTED** [review]

File: templates/data/compass-integration-with-dump-clean-data.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  - from: generic.tmpl
+    render:
+      - to: ../../prow/jobs/incubator/compass/post-main-compass-vm-clean-with-dump.yaml
+        localSets:
+          jobConfig_default:
+            path_alias: github.com/kyma-incubator/compass
+            command: "/home/prow/go/src/github.com/kyma-project/test-infra/prow/scripts/clean-vm-compass.sh"
+            args:
+              - "--vm-prefix"
+              - "compass-integration-test-with-dump"
```
[the remark this answers is not in the document]
Comment (dzahariev) [reply templates/data/compass-integration-with-dump-clean-data.yaml:11]: Exactly this I the reason I make it to accept parameters. To reuse the script in both jobs. 
If we make a script to work without parameters and to cleanup all VMs - one of the jobs will cleanup the other job VM and will kill the tests that are executed inside. Faster job will always finish and will kill the VM of both jobs.

File: prow/scripts/clean-vm-compass.sh [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+# shellcheck source=prow/scripts/lib/gcp.sh
+source "$SCRIPT_DIR/lib/gcp.sh"
+
+if [[ "${BUILD_TYPE}" == "pr" ]]; then
+    log::info "The job should not be executed on PRs"
+    exit 1
+elif [[ "${BUILD_TYPE}" == "master" ]] ; then
+    log::info "Main branch - cleanup should be done."
+    log::info "Execute Job Guard"
+    export JOB_NAME_PATTERN="(^pre-main-compass-integration$)|(^pre-main-compass-integration-no-dump$)"
```
[the remark this answers is not in the document]
Comment (dzahariev) [reply prow/scripts/clean-vm-compass.sh:26]: The post jobs are also scheduled with PJTester.

File: prow/scripts/provision-vm-compass.sh [hunk 5]
```diff
@@ -16,31 +16,70 @@ source "${SCRIPT_DIR}/lib/utils.sh"
...
 
-cleanup() {
-    ARG=$?
-    log::info "Removing instance compass-integration-test-${RANDOM_ID}"
-    gcloud compute instances delete --zone="${ZONE}" "compass-integration-test-${RANDOM_ID}" || true ### Workaround: not failing the job regardless of the vm deletion result
-    exit $ARG
-}
+# Log the VM recycle strategy
+if [[ "${RECYCLE_VM}" == "yes" && "${KYMA_INSTALLED}" == "yes" ]]; then
+    log::info "VM will be recycled and reused to execute validaiton of next commit in this PR: ${SUFFIX}"
```
[the remark this answers is not in the document]
Comment (dzahariev) [reply prow/scripts/provision-vm-compass.sh:75]: The job is executed either on PR or on main branch. The possible suffixes here are either "PR-XXXX" or "main".

**Review by PetarTodorovv — APPROVED** [review]

**Review by Ressetkk — APPROVED** [review]

**Pull request merged**